database/interface.py

```diff
@@ -211,37 +216,46 @@ def add_harvest_error(self, error_data: dict, error_type: str):
             self.db.rollback()
             return None
 
-    def get_all_errors_of_job(self, job_id: str) -> [[dict], [dict]]:
-        job = self.get_harvest_job(job_id)
-        job_errors = list(map(HarvesterDBInterface._to_dict, job.errors))
+    def add_harvest_record_error(self, error_data: dict):
```

Comment:
despite this function being almost identical to `add_harvest_job_error` they represent different things. i get the reason to do it.